### PR TITLE
osc: add typing dependency

### LINKED ISSUE #13230: Unable to run osc, getting import error for M2Crypto
# Please always follow these steps:
- [x] Confirmed this is a problem with only one, specific formula and not Homebrew/brew? If it's a general Homebrew/brew problem please file this issue at https://github.com/Homebrew/brew/issues/new
- [x] Ran `brew update` and retried your prior step?
- [x] Ran `brew doctor`, fixed as many issues as possible and retried your prior step?
- [x] Ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
https://gist.github.com/78345c2f2ffe85db0a839f9fbb21c793

To help us debug your issue please explain:
- What you were trying to do (and why)
Trying to run `osc`
- What happened (include command output)
```
cannot import name m2
M2Crypto is needed to access https://api.opensuse.org in a secure way.
Please install python-m2crypto.
```
- What you expected to happen
Success
- Step-by-step reproduction instructions (by running `brew install` commands)
```
brew install -s osc
osc ls
```

Also happens when I use the bottle.  *(Tried from source just because.)*

My PATH:
`PATH=/usr/local/sbin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/opt/X11/bin:/Applications/Wireshark.app/Contents/MacOS:/Users/MindTooth/.config/yarn/global/node_modules/.bin`

### PR DESCRIPTION
**Pull request by tdsmith**
Needed for M2Crypto.

Closes #13230.

### CODE REVIEW

**Review by ilovezfs — APPROVED** [review]

File: Formula/osc.rb [hunk 1]
```diff
@@ -35,6 +35,11 @@ class Osc < Formula
     sha256 "05d94fd9b2dae2fb8e072819a795f0e05d3611b09ea185f68e1630530ec09ae8"
   end
```
Comment (ilovezfs) [inline Formula/osc.rb]: Maybe add a comment so it doesn't get removed when not output by poet?

Comment (tdsmith) [inline Formula/osc.rb]: It doesn't?

Comment (tdsmith) [inline Formula/osc.rb]: If you install M2Crypto and run poet against it, you'll get the typing stanza, but getting either osc or M2Crypto installed is annoying. Do you remember getting output that lacked typing?

Comment (ilovezfs) [inline Formula/osc.rb]: Nope. Looks like it's never had it.

Comment (tdsmith) [inline Formula/osc.rb]: Nope to which?

Comment (ilovezfs) [inline Formula/osc.rb]: >Do you remember getting output that lacked typing?

To that.

The formula seems never to have had a `typing` stanza.

Comment (tdsmith) [inline Formula/osc.rb]: Okay! I think M2Crypto sprouted that recently.

**Pull request closed without merging**